platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -1231,9 +1051,8 @@ private void writeHashes(final long maxValidPath, final Stream<VirtualHashRecord
         });
 
         if (hasDiskStoreForHashes) {
-            final DataFileReader<VirtualHashRecord> newHashesFile = hashStoreDisk.endWriting();
-            statistics.setFlushHashesStoreFileSizeMb(
-                    newHashesFile == null ? 0 : newHashesFile.getSize() * BYTES_TO_MEBIBYTES);
+            statisticsUpdater.setFlushHashesStoreFileSize(hashStoreDisk.endWriting());
```

Comment:
Although it's correct and shorter than it used to be, it looks slightly less readable to me, it's easier to miss the call to `endWriting()` behind stat updates. My $0.02

Reply:
Agreed, I extracted a local variable as you suggested